api/repositories/pod_repository.go

```diff
@@ -0,0 +1,211 @@
+package repositories
+
+import (
+	"context"
+	"fmt"
+	"strconv"
+
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/labels"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+//+kubebuilder:rbac:groups=v1,resources=pods,verbs=get;list;watch
+//+kubebuilder:rbac:groups=v1,resources=pods/status,verbs=get
+
+const (
+	workloadsContainerName = "opi"
+	cfInstanceIndexKey     = "CF_INSTANCE_INDEX"
+	RunningState           = "RUNNING"
+	// All below statuses changed to "DOWN" until we decide what statuses we want to support in the future
+	pendingState = "STARTING"
+	crashedState = "DOWN"
+	unknownState = "DOWN"
+)
+
+type PodStatsRecord struct {
+	Type  string
+	Index int
+	State string `default:"DOWN"`
+}
+
+type FetchPodStatsMessage struct {
+	Namespace   string
+	AppGUID     string
+	Instances   int
+	ProcessType string
+}
+
+type PodRepo struct{}
+
+const (
+	AppGUIDKey = "cloudfoundry.org/app_guid" // TODO: Eirini currently uses this key - get them to update this?
+)
+
+func (r *PodRepo) FetchPodStatsByAppGUID(ctx context.Context, k8sClient client.Client, message FetchPodStatsMessage) ([]PodStatsRecord, error) {
+	labelSelector, err := labels.ValidatedSelectorFromSet(map[string]string{AppGUIDKey: message.AppGUID})
+	if err != nil {
+		return nil, err
+	}
+	listOpts := &client.ListOptions{Namespace: message.Namespace, LabelSelector: labelSelector}
+
+	pods, err := FetchPods(ctx, k8sClient, *listOpts)
+	if err != nil {
+		return nil, err
+	}
+
+	//Initialize records slice with the pod instances we expect to exist
+	records := make([]PodStatsRecord, message.Instances)
+	for i := 0; i < message.Instances; i++ {
+		records[i] = PodStatsRecord{
+			Type:  message.ProcessType,
+			Index: i,
+			State: "DOWN",
```

Comment:
Can we use the `unknownState` constant here?

Reply:
Addressed in commit https://github.com/cloudfoundry/cf-k8s-controllers/pull/248/commits/5a4d14ec897981ac8a15f7962ccf70c663466818